### PR TITLE
feat: update repo with latest changes from t3-turbo

### LINKED ISSUE #84: create-t3-turbo desync
**Describe the bug**

This repo is slowly getting out of sync with the "main" [create-t3-turbo](https://github.com/t3-oss/create-t3-turbo) repo. Is it worth to try to keep this one up to date? How can this repo be more in sync? Would a guide to replacing NextAuth for Clerk be better?

### PR DESCRIPTION
**Pull request by afonsojramos**
Closes #84.

From local testing everything seems to be working, but it might need some polishing.

### CODE REVIEW

File: apps/nextjs/src/_middleware.ts
[the remark this answers is not in the document]
Comment (afonsojramos) [reply apps/nextjs/src/_middleware.ts:1]: Why not, updated! Did you find anything else that was not working as expected?

**Comment by afonsojramos** [comment]
Hey @perkinsjr, can you also take a look?

**Comment by perkinsjr** [comment]
I can but the PR is huge so it's going to take me a way to review all the code, functionality and production deployments

**Comment by afonsojramos** [comment]
Totally understandable! Most code came directly from the https://github.com/t3-oss/create-t3-turbo, I've just did a sort merge from both.

**Comment by afonsojramos** [comment]
Now that I think of it, maybe we can actually not have renovate in this repo, and we'll just cherry-pick commits from the OG repo... Not sure 🤷

**Comment by afonsojramos** [comment]
I've given up on this PR. This repo should have been forked and not generated from https://github.com/clerkinc/t3-turbo-and-clerk just like the Supabase fork (https://github.com/supabase-community/create-t3-turbo), where we can see the changes that enabled Supabase Auth (https://github.com/supabase-community/create-t3-turbo/compare/t3-oss:create-t3-turbo:main...main) whereas in this one we need to guess.

**Pull request closed without merging**